### PR TITLE
feat: add product statuses to result component

### PR DESCRIPTION
**Pull request by tuanddd**

### CODE REVIEW

File: packages/search-ui/src/Results/components/Result/styles.ts [hunk 1]
```diff
@@ -21,14 +23,30 @@ export default function useResultStyles(props: UseResultStylesParams) {
     description: [tw`mt-1 text-sm text-gray-500`],
     rating: [tw`mt-1 space-y-1 text-center`],
     priceContainer: [],
-    price: [isOnSale ? tw`font-medium text-red-500` : ''],
```
Comment (zlatanpham) [inline packages/search-ui/src/Results/components/Result/styles.ts:24]: Not sure if we still need `font-medium` for `onSale`.

File: packages/search-ui/src/ContextProvider/types.ts [hunk 2]
```diff
@@ -28,6 +28,9 @@ export interface SearchUIContextProviderValues {
       searchingMessage?: string;
       emptyMessage?: string;
       errorMessage?: string;
+      onSaleStatus?: string;
+      outOfStockStatus?: string;
+      newArrivalStatus?: string;
```
Comment (zlatanpham) [inline packages/search-ui/src/ContextProvider/types.ts:33]: Looks like we need to update the docs for the change.

File: packages/search-ui/src/Results/components/Result/index.tsx [hunk 3]
```diff
@@ -87,7 +95,31 @@ const Result = React.memo(
...
-    const styles = getStylesObject(useResultStyles({ ...props, appearance, isOnSale }), disableDefaultStyles);
+    const isOutOfStock = React.useMemo(() => {
+      if (isEmpty(quantity)) {
+        return false;
+      }
+      const parseQuantities = (input: string | Array<string>) => (isArray(input) ? input : [input]).map(Number);
+      const quantities = parseQuantities(quantity);
+
+      return quantities[activeImageIndex] === 0;
+    }, [JSON.stringify(quantity), activeImageIndex]);
```
Comment (zlatanpham) [inline packages/search-ui/src/Results/components/Result/index.tsx:106]: Why do we need to `JSON.stringify`? Given we've already wrapped the `Result` component inside `React.memo` with a similar check.

File: packages/search-ui/src/Results/components/Result/styles.ts (hunk 1 above)
Comment (tuanddd) [reply packages/search-ui/src/Results/components/Result/styles.ts:24]: If so I think `font-medium` should be applied to all statuses and not just `onSale` for consistency sake.

File: packages/search-ui/src/ContextProvider/types.ts (hunk 2 above)
Comment (tuanddd) [reply packages/search-ui/src/ContextProvider/types.ts:33]: Updated

File: packages/search-ui/src/Results/components/Result/index.tsx (hunk 3 above)
Comment (tuanddd) [reply packages/search-ui/src/Results/components/Result/index.tsx]: If might as well remove the `JSON.stringify` call at `isOnSale` memo as well?

**Comment by tuanddd** [comment]
@zlatanpham I've also added the translation for all languages (only english is correct, others need to be updated)

File: packages/search-ui/src/Results/components/Result/index.tsx (hunk 3 above)
Comment (zlatanpham) [reply packages/search-ui/src/Results/components/Result/index.tsx]: I would say so

**Review by zlatanpham — APPROVED** [review]
LGTM 👍

**Pull request merged**